toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -105,113 +110,119 @@ func PrintBuildSummary(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, bu
-	logger.Log.Infof("Number of tested SRPMs:            %d", len(testedSRPMs))
-	logger.Log.Infof("Number of prebuilt SRPMs:          %d", len(prebuiltSRPMs))
-	logger.Log.Infof("Number of prebuilt delta SRPMs:    %d", len(prebuiltDeltaSRPMs))
-	logger.Log.Infof("Number of skipped SRPMs tests:     %d", len(skippedSRPMsTests))
-	logger.Log.Infof("Number of failed SRPMs:            %d", len(failedSRPMs))
-	logger.Log.Infof("Number of failed SRPMs tests:      %d", len(failedSRPMsTests))
-	logger.Log.Infof("Number of blocked SRPMs:           %d", len(blockedSRPMs))
-	logger.Log.Infof("Number of blocked SRPMs tests:     %d", len(blockedSRPMsTests))
-	logger.Log.Infof("Number of unresolved dependencies: %d", len(unresolvedDependencies))
-
-	if allowToolchainRebuilds && (len(rpmConflicts) > 0 || len(srpmConflicts) > 0) {
-		logger.Log.Infof("Toolchain RPMs conflicts are ignored since ALLOW_TOOLCHAIN_REBUILDS=y")
-	}
-
-	if len(rpmConflicts) > 0 || len(srpmConflicts) > 0 {
-		conflictsLogger("Number of toolchain RPM conflicts: %d", len(rpmConflicts))
-		conflictsLogger("Number of toolchain SRPM conflicts: %d", len(srpmConflicts))
-	}
-
-	if len(builtSRPMs) != 0 {
-		logger.Log.Info("Built SRPMs:")
-		for srpm := range builtSRPMs {
-			logger.Log.Infof("--> %s", filepath.Base(srpm))
-		}
-	}
-
-	if len(testedSRPMs) != 0 {
-		logger.Log.Info("Tested SRPMs:")
-		for srpm := range testedSRPMs {
-			logger.Log.Infof("--> %s", filepath.Base(srpm))
-		}
-	}
+	// print summary
+	printSummary(failedSRPMs, failedSRPMsTests, prebuiltSRPMs, prebuiltDeltaSRPMs, builtSRPMs, testedSRPMs, skippedSRPMsTests, unresolvedDependencies, blockedSRPMs, blockedSRPMsTests, rpmConflicts, srpmConflicts, allowToolchainRebuilds, conflictsLogger)
 
 	if len(prebuiltSRPMs) != 0 {
-		logger.Log.Info("Prebuilt SRPMs:")
-		for srpm := range prebuiltSRPMs {
-			logger.Log.Infof("--> %s", filepath.Base(srpm))
+		logger.Log.Info(color.GreenString("Prebuilt SRPMs:"))
+		keys := sliceutils.MapToSlice(prebuiltSRPMs)
+		sort.Strings(keys)
+		for _, prebuiltSRPM := range keys {
+			logger.Log.Infof("--> %s", filepath.Base(prebuiltSRPM))
 		}
 	}
 
 	if len(prebuiltDeltaSRPMs) != 0 {
-		logger.Log.Info("Skipped SRPMs (i.e., delta mode is on, packages are already available in a repo):")
-		for srpm := range prebuiltDeltaSRPMs {
-			logger.Log.Infof("--> %s", filepath.Base(srpm))
+		logger.Log.Info(color.GreenString("Skipped SRPMs (i.e., delta mode is on, packages are already available in a repo):"))
+		keys := sliceutils.MapToSlice(prebuiltDeltaSRPMs)
+		sort.Strings(keys)
+		for _, prebuiltDeltaSRPM := range keys {
+			logger.Log.Infof("--> %s", filepath.Base(prebuiltDeltaSRPM))
 		}
 	}
 
 	if len(skippedSRPMsTests) != 0 {
-		logger.Log.Info("Skipped SRPMs tests:")
-		for srpm := range skippedSRPMsTests {
-			logger.Log.Infof("--> %s", filepath.Base(srpm))
+		logger.Log.Info(color.RedString("Skipped SRPMs tests:"))
+		keys := sliceutils.MapToSlice(skippedSRPMsTests)
+		sort.Strings(keys)
```

Comment:
Seeing how often you call `MapToSlice` followed by `sort.Strings`, I'd suggest adding another function to avoid this repetition. For instance `sliceutils.MapToSortedSlice`.